server/src/main/java/com/linecorp/centraldogma/server/internal/api/ContentServiceV1.java

```diff
@@ -441,4 +443,20 @@ public static void checkMirrorLocalRepo(String repoName, Iterable<Change<?>> cha
             }
         }
     }
+
+    /**
+     * Checks if the commit is for creating a file and raises a {@link RepositoryNotAllowedException} if the
+     * given {@code repoName} field is one of {@code meta} and {@code dogma} which are internal repositories.
+     */
+    public static void checkPushLocalRepo(String repoName, Iterable<Change<?>> changes) {
+        if (Project.isReservedRepoName(repoName)) {
+            final boolean hasChangesWithoutMirroring =
+                    Streams.stream(changes)
+                           .anyMatch(change -> !DefaultMetaRepository.PATH_MIRRORS.equals(change.path()));
+            if (hasChangesWithoutMirroring) {
+                throw new RepositoryNotAllowedException(
```

Comment:
Maybe `InvalidPushException`?

Reply:
Nice idea.
If so, which HTTP status do you expect when `InvalidPushException` is raised? `400 Bad Request` of `403 Forbidden`?

Reply:
I would go for 403. 😄

Reply:
I think we can also throw `InvalidPushException` in `checkMirrorLocalRepo`. 😉